src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValueOfJsonPrimitive.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization.Metadata;
+
+namespace System.Text.Json.Nodes
+{
+    internal static class JsonValueOfJsonPrimitive
+    {
+        internal static JsonValue CreatePrimitiveValue(ref Utf8JsonReader reader, JsonSerializerOptions options)
+        {
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.False:
+                case JsonTokenType.True:
+                    return new JsonValueOfJsonBool(reader.GetBoolean(), options);
+                case JsonTokenType.String:
+                    byte[] quotedStringValue;
+
+                    if (reader.HasValueSequence)
+                    {
+                        // Throw if the long to int conversion fails. This can be accommodated in the future if needed by
+                        // adding another derived class that has a backing ReadOnlySequence<byte> instead of a byte[].
+                        int bufferLength = checked((int)(2 + reader.ValueSequence.Length));
+                        quotedStringValue = new byte[bufferLength];
+                        reader.ValueSequence.CopyTo(quotedStringValue.AsSpan().Slice(1, bufferLength - 2));
+                    }
+                    else
+                    {
+                        quotedStringValue = new byte[2 + reader.ValueSpan.Length];
+                        reader.ValueSpan.CopyTo(quotedStringValue.AsSpan(1, reader.ValueSpan.Length));
+                    }
+
+                    quotedStringValue[0] = quotedStringValue[quotedStringValue.Length - 1] = JsonConstants.Quote;
+                    return new JsonValueOfJsonString(quotedStringValue, reader.ValueIsEscaped, options);
+                case JsonTokenType.Number:
+                    byte[] numberValue = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan.ToArray();
+                    return new JsonValueOfJsonNumber(new JsonValueOfJsonNumber.JsonNumber(numberValue), options);
+                default:
+                    Debug.Fail("Only primitives allowed.");
+                    ThrowHelper.ThrowJsonException();
+                    return null!; // Unreachable, but required for compilation.
+            }
+        }
+
+        private sealed class JsonValueOfJsonString : JsonValue
+        {
+            private readonly byte[] _value;
```

Comment:
What's the benefit of storing a potentially unescaped UTF-8 string with quotes when this could just wrap a regular string that has been fully unescaped?

Reply:
There is a one trade-off: if the user calls `TryGetValue<JsonElement>()` then we need to escape the string before we call `JsonElement.Parse`. For all other types (`string`, `char`, `Guide`, `DateTime`, and `DateTimeOffset`) it would be better to unescape. I left it as escaped to keep the logic closer to JsonValueOfElement which assumes the underlying value can be escaped.

Reply:
How common is it for users to want to call `TryGetValue<JsonElement>()`? My understanding is that the canonical use case is converting it to primitives, so perhaps it's better to optimize for that use case?

Reply:
Agree, updated